### PR TITLE
Add endpoint to retrieve user organizations on github

### PR DESCRIPTION
**Pull request by denis-yuen**
Working on https://github.com/dockstore/dockstore/issues/2671

Edit: can split out ticket to make SourceControlResourceInterface.getAndRefreshTokens not check on all source control tokens, but just check on the token for the source control you actually wanted ... probably not a priority though

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]
I suggest that it should return an error if the user doesn't have a GitHub token or if the `getMyOrganizations` call fails, e.g., due to a bad token. Now it will just return an empty array in both cases, and the user won't know that it failed behind the scenes.

**Review by garyluu — COMMENTED** [review]
Can omit the swagger.yaml and flesh out the openapi.yaml (response code, response description)

**Comment by denis-yuen** [comment]
> I suggest that it should return an error if the user doesn't have a GitHub token or if the `getMyOrganizations` call fails, e.g., due to a bad token. Now it will just return an empty array in both cases, and the user won't know that it failed behind the scenes.

Actually, think we get this for free already since 
https://github.com/dockstore/dockstore/blob/1.11.0-alpha.1/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java#L50 should run into https://github.com/dockstore/dockstore/blob/1.11.0-alpha.1/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java#L319

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 1]
```diff
@@ -941,6 +944,17 @@ public SourceFile getSourceFile(String path, String id, String branch, Descripto
         throw new UnsupportedOperationException("not implemented/needed for github");
     }
 
+    @Override
+    public List<SourceControlOrganization> getOrganizations() {
+        try {
+            return github.getMyOrganizations().entrySet().stream()
+                    .map(o -> new SourceControlOrganization(o.getValue().getId(), o.getKey())).collect(Collectors.toList());
+        } catch (IOException e) {
+            LOG.info(githubTokenUsername + ": Cannot retrieve their organizations", e);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:953]: > Actually, think we get this for free already since...

That's true, but here it's swallowing IOException (not sure when they will occur), and if there is not a GitHub token associated with the account, it returns an empty array list on line 955. Both are sort of edge cases, since almost all our users have GitHub accounts.

**Review by coverbeck — APPROVED** [review]

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java [hunk 2]
```diff
@@ -546,6 +547,21 @@ public void testUpdateUserMetadataFromGithub() {
...
+        io.dockstore.openapi.client.api.UsersApi userApi = new io.dockstore.openapi.client.api.UsersApi(userWebClient);
+        List<SourceControlOrganization> myGitHubOrgs = userApi.getMyGitHubOrgs();
+        assertTrue(!myGitHubOrgs.isEmpty() && myGitHubOrgs.stream().anyMatch(org -> org.getName().equals("dockstoretesting")));
+        // Delete all of the tokens (except for Dockstore tokens) for every user
+        testingPostgres.runUpdateStatement("UPDATE token set content = 'foo' WHERE tokensource <> 'dockstore'");
+
+        try {
+            userApi.getMyGitHubOrgs();
+        } catch (io.dockstore.openapi.client.ApiException e) {
+            assertEquals(e.getCode(), HttpStatus.SC_BAD_REQUEST);
```
Comment (garyluu) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java:561]: I think these parameters are swapped

**Review by garyluu — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java (hunk 1 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:953]: I don't think they can get here. A bad token should not result in a IOException, the check occurs under https://github.com/dockstore/dockstore/pull/4171/files#diff-2c8f98ad7fbc26776c9437aeb9dfe12d07fe60de2ada259bf78564e92198295aR811 which is a CustomWebApplicationException which isn't related to IOException. That's why the test is getting a 400 and not a 500? (not sure what a IOException works out to)

Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:953]: Oh, I see, you mentioned two cases in the original comment, bad token and no token. 

No token is a bit funny. A user that intentionally un-linked their github account or logged in through Google. Hmm

**Review by NatalieEO — APPROVED** [review]
